### DISCUSSION TITLE
Receiving a warning when not passing a ref with controlled component

### DISCUSSION THREAD

**Discussion by ahmedosama7450** [react-hook-form/react-hook-form#5085] [Q&A] [answered]
I am trying to integrate react hook form with [Headless UI Select Component](https://headlessui.dev/react/listbox)
This component is built from scratch, It doesn't have an underlying native html select tag, I guess. So, I don't have access to a `ref`. It only exposes `onChange` and `value` ([API Reference](https://headlessui.dev/react/listbox#component-api)).

I am using the hook form `Controller` to wrap a custom select like this
```
<Controller
  name='year'
  control={control}
  render={({ field }) => {
    return (
      <CustomSelect
        value={field.value}
        onChange={(value)=> field.onChange(value)}
        label='Year'
        className='mt-5 sm:flex-1'
        options={[
          {
            value: 1,
            text: 'First Year',
          },
          {
            value: 2,
            text: 'Second Year',
          },
        ]}
      />
    );
  }}
/>
```
I think everything is working just fine except that I receive this warning in chrome devtools
```
Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?

Check the render method of `Controller`.
    at CustomSelect (http://localhost:3000/_next/static/chunks/pages/create-profile.js?ts=1620420857237:473:22)
    at Controller (http://localhost:3000/_next/static/chunks/pages/create-profile.js?ts=1620420857237:40059:37)
    at form
    at div
    at div
    at main
    at div
    at CreateProfile (http://localhost:3000/_next/static/chunks/pages/create-profile.js?ts=1620420857237:46240:81)
    ....
```

I think that you require passing `ref` in controlled components for things like focusing the component. I don't really need this and also this component doesn't support that, I guess.

So, Is it safe not to pass `ref` with `Controller` component ? If so, How do I suppress this warning ?

**Comment by leapful** [comment, marked as answer]
@ahmedosama7450 Please provide a codesandbox to get more detail about the issue.

Usually, the warning about "React.forwardRef" only happens when attribute "ref" is added to a functional component which is not handling "forwardRef" so the warning might be caused by somewhere else inside your "CustomSelect" that is using "ref".

**Reply by ahmedosama7450** [reply]
Yes, you're right. I was spreading the controller `field` into the component
```
<Controller
  name='year'
  control={control}
  render={({ field }) => {
    return (
      <CustomSelect
        {...field}
        label='Year'
        className='mt-5 sm:flex-1'
        options={[
          {
            value: 1,
            text: 'First Year',
          },
          {
            value: 2,
            text: 'Second Year',
          },
        ]}
      />
    );
  }}
/>
```
After manually passing `value` and `onChange`, the warning is gone.

So, Is it safe not to pass `ref` with controlled components ?

**Reply by leapful** [reply]
@ahmedosama7450 

When using Controller component, you don't need to pass "ref", the prop "field" from "render" of Controller component will handle change events and value.
